File: leetcode/exercises/1.1.1.easy.contains_duplicate_2.py
# -----------------------
# Created     : 23/10/2025
# Last Edited : 23/10/2025
# Topics      : 
# Big O       : 
# Problem Id  : 217
# Source      :
# Notes       :
# -----------------------

class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        nums.sort()
        for i in range(1, len(nums)):
            if nums[i-1] == nums[i]:
                return True
        return False

sol = Solution()
print()
print(sol.containsDuplicate([2, 1, 10, 12, 1]))

File: leetcode/exercises/1.1.1.easy.contains_duplicate_2.go
// -----------------------
// Created     : 30/10/2025
// Last Edited : 30/10/2025
// Topics      : 
// Big O       :
// Problem Id  : 217
// Source      :
// Notes       :
// -----------------------

package main

import "fmt"
import "sort"

func containsDuplicate(nums []int) bool {
    sort.Ints(nums)
    for i := 1; i < len(nums); i++ {
        if nums[i-1] == nums[i] { return true }
    }
    return false
}

func main() {
    fmt.Println(containsDuplicate([]int{5, 2, 3, 10, 8, 1}))
}
